### PR TITLE
Do not publish docker release images on `-dev` tags

### LINKED ISSUE #25193: latest tag being bind to the wrong Docker image
### Description

The `latest` tag of your [Docker image](https://hub.docker.com/r/gitea/gitea/tags) is often bind to the wrong Gitea version. I often notice that it's bind to some RC version, dev version, or currently to the `1.21.0-dev` version, which is definitely not good.

I notice it because I'm running Watchtower daily to always keep my containers up-to-date, and right now my Gitea container has been bumped to this dev version and is no longer working because of an undocumented breaking change.

I can also think about the people attempting to casually download and run Gitea for the first time with a simple `gitea/gitea` and won't understand why it's not working.

So can you please make sure the `:latest` tag **always** points to the latest **stable** version please?

### Gitea Version

1.21.0

### Can you reproduce the bug on the Gitea demo site?

No

### Log Gist

_No response_

### Screenshots

latest
---

### Git Version

_No response_

### Operating System

Raspberry Pi OS 64bit

### How are you running Gitea?

:latest Docker image

### Database

SQLite

### PR DESCRIPTION
**Pull request by delvh**
Try to prevent what happened with tag `v1.21.0-dev` as outlined in #25193.
Unfortunately, we cannot really test if it works as intended as we would need to release a new `dev` tag for that.
Fixes #25193 (or at least attempts to).

### CODE REVIEW

**Review by silverwind — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Comment by jolheiser** [comment]
I don't think `raymond` understands `or`.

**Comment by jolheiser** [comment]
https://go.dev/play/p/Hg-GtByHgri

Even worse, it doesn't even choke, it just happily accepts random input.
https://go.dev/play/p/no6T_yrD5JC

**Comment by delvh** [comment]
I don't know if it does or doesn't.
As I said, I can't test it, so this PR is a shot in the dark.
If it doesn't support that, then how exactly are we supposed to check for two conditions?

**Comment by jolheiser** [comment]
> If it doesn't support that, then how exactly are we supposed to check for two conditions?

Apparently nesting :frowning_face: 
https://go.dev/play/p/LXy1klCnwXz

File: docker/manifest.rootless.tmpl [hunk 1]
```diff
@@ -1,12 +1,14 @@
 image: gitea/gitea:{{#if build.tag}}{{trimPrefix "v" build.tag}}{{else}}{{#if (hasPrefix "refs/heads/release/v" build.ref)}}{{trimPrefix "refs/heads/release/v" build.ref}}-{{/if}}nightly{{/if}}-rootless
 {{#if build.tags}}
-{{#unless (contains "-rc" build.tag)}}
+{{#unless (contains "-rc" tag)}}
+{{#unless (contains "-dev" tag)}}
```
Comment (jolheiser) [inline docker/manifest.rootless.tmpl]:
```suggestion
{{#unless (contains "-rc" build.tag)}}
{{#unless (contains "-dev" build.tag)}}
```

File: docker/manifest.tmpl [hunk 2]
```diff
@@ -1,12 +1,14 @@
 image: gitea/gitea:{{#if build.tag}}{{trimPrefix "v" build.tag}}{{else}}{{#if (hasPrefix "refs/heads/release/v" build.ref)}}{{trimPrefix "refs/heads/release/v" build.ref}}-{{/if}}nightly{{/if}}
 {{#if build.tags}}
-{{#unless (contains "-rc" build.tag)}}
+{{#unless (contains "-rc" tag)}}
+{{#unless (contains "-dev" tag)}}
```
Comment (jolheiser) [inline docker/manifest.tmpl]:
```suggestion
{{#unless (contains "-rc" build.tag)}}
{{#unless (contains "-dev" build.tag)}}
```

**Review by jolheiser — COMMENTED** [review]
My bad, I had shortened the variables in my examples as I wrote them hastily :sweat_smile:

**Review by jolheiser — APPROVED** [review]

**Pull request merged**